app/assets/javascripts/run_in_jupyter.js.erb

```diff
@@ -10,15 +10,28 @@ $(document).ready(function() {
   %>
   <%="var jupyter_interfaces=" + interfaces.to_json%>
 
-  // Clearning out turoblinks duplicated bindings
-  $(document).off('click','.run_jupyter');
+  $(document).on('click','#environmentSelectedFormSubmit', function(e) {
+    e.preventDefault();
+    var environment = $("input[name='selectedEnvironment']:checked").val();
+    var temp = environment.split('--');
+    var url = temp[0];
+    var user_interface = temp[1];
+    get_notebook(url, $('#runInJupyterId').val(), $('#runInJupyterType').val(), $('#runInJupyterRef').val(), user_interface);
+    $('#runInJupyterModal').modal('hide');
+  });
 
-  $(document).on('click','.run_jupyter', function() {
-    bootbox.dialog({
-      title: 'Opening ...',
-      message: '<div class="progress progress-striped active"><div class="progress-bar"  role="progressbar" aria-valuenow="45" aria-valuemin="0" aria-valuemax="100" style="width: 100%">'
-    });
+  $(document).on('click','#runInJupyterOverwriteSubmit', function(e) {
+    e.preventDefault();
+    var name =  $('#runInJupyterName').val();
+    var url =  $('#runInJupyterUrl').val();
+    var notebook = $('#runInJupyterNotebook').val();
+    var user_interface =  $('#runInJupyterUserInterface').val();
+    send(name,url,notebook,user_interface);
+    $('#runInJupyterOverwrite').modal('hide');
+  });
 
+  $(document).on('click','.run_jupyter', function() {
+    loadingGif();
```

Comment:
This won't work on most pages yet.  Might roll back to the dialog for now, or just skip having a "opening" dialog at all, and then come back and do loadingGif here later maybe?